tensoradapter/pytorch/torch.cpp

```diff
@@ -5,57 +5,31 @@
  */
 
 #include <tensoradapter_exports.h>
-#include <torch/torch.h>
-#include <ATen/DLConvertor.h>
+#include <c10/core/CPUAllocator.h>
 #ifdef DGL_USE_CUDA
 #include <c10/cuda/CUDACachingAllocator.h>
+#include <cuda_runtime.h>
 #endif  // DGL_USE_CUDA
-#include <vector>
-#include <iostream>
-
-#if DLPACK_VERSION > 040
-// Compatibility across DLPack - note that this assumes that the ABI stays the same.
-#define kDLGPU kDLCUDA
-#define DLContext DLDevice
-#endif
 
 namespace tensoradapter {
 
-static at::Device get_device(DLContext ctx) {
-  switch (ctx.device_type) {
-   case kDLCPU:
-    return at::Device(torch::kCPU);
-    break;
-   case kDLGPU:
-    return at::Device(torch::kCUDA, ctx.device_id);
-    break;
-   default:
-    // fallback to CPU
-    return at::Device(torch::kCPU);
-    break;
-  }
-}
-
 extern "C" {
 
-TA_EXPORTS DLManagedTensor* TAempty(
```

Comment:
Instead of deprecating `TAempty`, can we still use torch current stream for `tensor.empty()` but apply stream sync (with DGL stream) inside this function?

Reply:
Then we have to call `record_stream` to allow the tensor being used in another stream.

Deprecating `TAempty` also helps us get rid of the DLPack conversion.